core/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -120,8 +121,11 @@ protected boolean validate(ClusterState newState) {
         AsyncSingleAction(Task task, Request request, ActionListener<Response> listener) {
             this.task = task;
             this.request = request;
-            if (task != null) {
-                request.setParentTask(clusterService.localNode().getId(), task.getId());
+            DiscoveryNode discoveryNode = clusterService.localNode();
+            if (task != null && discoveryNode != null) {
```

Comment:
These clauses can not be combined in a conjunction like this because now you're throwing when `task` is null. Maybe we should throw when `task` is null, I don't know, but let's not change the current behavior (and the exception message does not fit that case anyway
).